projects/plugins/jetpack/extensions/blocks/one-payment/variations.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * The different payment blocks that can be chosen.
+ */
+
+/**
+ * External dependancies
+ */
+
+import { getBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { settings as DonationSettings } from '../donations';
+import { settings as RecurringPaymentSettings } from '../recurring-payments';
+
+const variations = [
+	[ 'donations', DonationSettings ],
+	[ 'recurring-payments', RecurringPaymentSettings ],
+];
+
+const variationDefinitions = variations
+	.map( blockNameSettings => {
+		const blockDefinition = {};
+		const [ blockName, settings ] = blockNameSettings;
+		if ( getBlockType( 'jetpack/' + blockName ) ) {
+			blockDefinition.name = 'jetpack/onepayment-' + blockName;
+			blockDefinition.title = settings.title;
+			blockDefinition.description = settings.description;
+			blockDefinition.icon = settings.icon.src;
+			blockDefinition.innerBlocks = [ [ 'jetpack/' + blockName, {} ] ];
+			// The inner block itself is already listed in the inserter in its own right, so just include in this blocks
+			// unified intro.
+			blockDefinition.scope = [ 'block' ];
```

Comment:
Bit of a nitpick, but I'm personally not a fan of declaring a const and then changing its properties. Seems a bit like cheating to me. 😄 

We can avoid declaring `blockDefinition` at the beginning of the map, and early return instead.
E.g.
```js
const [ blockName, settings ] = blockNameSettings;
if ( getBlockType( 'jetpack/' + blockName ) ) {
  return {
    name: 'jetpack/onepayment-' + blockName,
    title: settings.title,
    /* ... */
  };
}
return {};
```

Reply:
It's so not cheating! It's part of the language, it's always referencing the same thing. Using a const prevents reassigning things to that symbol and so prevents bugs.

That said your suggestion is neater in this case, thank-you 👍

Reply:
Ahah, yes, I know it's not cheating, and I understand the logic behind it, but still, my brain parses it as such. 😄 

Jokes and brain defects aside, I've noticed that some of these rules (self-imposed like this, or from strict linting, etc.) often push me to rethink the approach, and eventually, the code gets leaner or more elegant.
(TBF there's a bit of bias here as well, as I'm super into early returns. 😄 )